src/main/java/com/synopsys/integration/alert/web/controller/handler/SystemHandler.java

```diff
@@ -74,4 +78,33 @@ public SystemHandler(final ContentConverter contentConverter, final SystemAction
             return createResponse(HttpStatus.BAD_REQUEST, ex.getMessage());
         }
     }
+
+    public ResponseEntity<String> getCurrentSetup(final String contextPath) {
+        if (actions.isSystemInitialized()) {
+            final HttpHeaders headers = new HttpHeaders();
+            headers.add("Location", contextPath);
+            return new ResponseEntity<>(headers, HttpStatus.FOUND);
+        } else {
+            return new ResponseEntity<>(getContentConverter().getJsonString(actions.getCurrentSystemSetup()), HttpStatus.OK);
+        }
+    }
+
+    public ResponseEntity<String> saveRequiredInformation(final SystemSetupModel requiredSystemConfiguration) {
+        final ResponseEntity<String> response;
+        if (actions.isSystemInitialized()) {
+            response = new ResponseEntity<>(HttpStatus.METHOD_NOT_ALLOWED);
```

Comment:
Gavin and I were just talking about something similar to this. Should we respond with METHOD_NOT_ALLOWED, when the type of method is actually allowed, but just not all criteria is met?